cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -122,6 +131,30 @@ func (s *server) addArchiveStorage(opts *querysvc.QueryServiceOptions, host comp
 	return nil
 }
 
+func (s *server) createMetricStorage(host component.Host) (metricsstore.Reader, error) {
+	if s.config.MetricStorage == "" {
+		s.telset.Logger.Info("Metric storage not configured")
+		return disabled.NewMetricsReader()
+	}
+
+	mf, err := jaegerstorage.GetMetricsFactory(s.config.MetricStorage, host)
+	if err != nil {
+		return nil, fmt.Errorf("cannot find metrics storage factory: %w", err)
+	}
+
+	// metricsReader, ok := opts.InitMetricStorage(mf, s.telset.Logger)
+	metricsReader, err := mf.CreateMetricsReader()
+	if errors.Is(err, storage.ErrMetricStorageNotConfigured) || errors.Is(err, storage.ErrMetricStorageNotSupported) {
+		s.telset.Logger.Info("Metric storage not created", zap.String("reason", err.Error()))
+		return disabled.NewMetricsReader()
+	}
+	if err != nil {
+		s.telset.Logger.Error("Cannot init metric storage reader", zap.Error(err))
+		return disabled.NewMetricsReader()
```

Comment:
why not bubble up the error in this case? You already handle soft errors in the previous `if`